lib/mixlib/install/options.rb

```diff
@@ -71,8 +79,7 @@ def validate_channels
       end
 
       def validate_unstable_version
-        if ARTIFACTORY_CHANNELS.include?(channel) &&
-            product_version !~ /^\d+.\d+.\d+\+[0-9]{14}$/
+        if for_artifactory? && product_version !~ /^\d+.\d+.\d+\+[0-9]{14}$/
```

Comment:
We added this because we were running into issues where we get multiple candidates from artifactory when you specify a version like `12.5.1` only. I recommend we merge with this logic and do the necessary when we are doing the minor changes when we enable :unstable channel in the artifactory for reals.